cmd/thanos/query.go

```diff
@@ -91,6 +91,8 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 	enablePartialResponse := cmd.Flag("query.partial-response", "Enable partial response for queries if no partial_response param is specified.").
 		Default("true").Bool()
 
+	storeReadTimeout := modelDuration(cmd.Flag("store.read-timeout", "Maximum time to wait for any data from store. If Store doesn't send any data in storeReadTimeout the Store will be ignored and partial data will be returned. 0 disables timeout.").Default("0ms"))
```

Comment:
I believe this is either `grpc-client-timeout` or `store.response-timeut` . `Read` relates to much to the `HTTPConfig` `ReadTimeout` in Golang. (: Thoughts?

Reply:
I agree with that observation, and `storeReadTimeout` is not visible to users so it might raise some confusion. I suggest something like: `If a Store doesn't send any data in this specified duration then a Store will be ignored and partial data will be returned`.

Reply:
@GiedriusS +1